src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -324,19 +379,30 @@ SAssetBundle CGraphicsPipelineLoaderMTL::loadAsset(io::IReadFile* _file, const I
 
         core::smart_refctd_ptr<ICPUDescriptorSet> ds3;
         {
-            const std::string dsCacheKey = std::string(fullName.c_str())+"?"+materials[i].name+"?_ds";
-            const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_DESCRIPTOR_SET, (asset::IAsset::E_TYPE)0u };
-            auto ds_bundle = _override->findCachedAsset(dsCacheKey, types, ctx.inner, _hierarchyLevel + ICPUMesh::DESC_SET_HIERARCHYLEVELS_BELOW);
-            if (!ds_bundle.isEmpty())
-                ds3 = core::smart_refctd_ptr_static_cast<ICPUDescriptorSet>(ds_bundle.getContents().begin()[0]);
-            else 
+                const std::string dsCacheKey = std::string(fullName.c_str()) + "?" + materials[i].name + "?_ds";
+            if (_override)
             {
-                auto views = loadImages(relPath.c_str(), materials[i], ctx);
-                ds3 = makeDescSet(std::move(views), layout->getDescriptorSetLayout(3u));
-                if (ds3)
+                const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_DESCRIPTOR_SET, (asset::IAsset::E_TYPE)0u };
+                auto ds_bundle = _override->findCachedAsset(dsCacheKey, types, ctx.inner, _hierarchyLevel + ICPUMesh::DESC_SET_HIERARCHYLEVELS_BELOW);
+                if (!ds_bundle.isEmpty())
+                    ds3 = core::smart_refctd_ptr_static_cast<ICPUDescriptorSet>(ds_bundle.getContents().begin()[0]);
+                else
                 {
-                    SAssetBundle bundle{ds3};
-                    _override->insertAssetIntoCache(bundle, dsCacheKey, ctx.inner, _hierarchyLevel + ICPURenderpassIndependentPipeline::DESC_SET_HIERARCHYLEVELS_BELOW);
+                    auto views = loadImages(relPath.c_str(), materials[i], ctx);
+                    ds3 = makeDescSet(std::move(views), layout->getDescriptorSetLayout(3u));
+                    if (ds3)
+                    {
+                        SAssetBundle bundle{ ds3 };
+                        _override->insertAssetIntoCache(bundle, dsCacheKey, ctx.inner, _hierarchyLevel + ICPURenderpassIndependentPipeline::DESC_SET_HIERARCHYLEVELS_BELOW);
+                    }
+                }
+            }
+            else {
+                const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_DESCRIPTOR_SET, (asset::IAsset::E_TYPE)0u };
+                auto b = m_assetMgr->findAssets(dsCacheKey, types);
+                if (b)
+                {
+                    ds3 = core::smart_refctd_ptr_dynamic_cast<ICPUDescriptorSet>(b->begin()->getContents().begin()[0]);
```

Comment:
when using loadAsset() from the constructor, _override is equal to null ptr, and only then. The plan is to load the cached descriptor set with the magenta checkerboard and use it.